lib/solr/configuration.rb

```diff
@@ -1,32 +1,64 @@
-require 'solr/field_configuration/dynamic_field'
-require 'solr/field_configuration/field'
-require 'solr/field_configuration/field_definition_builder'
+require 'solr/core_configuration/dynamic_field'
+require 'solr/core_configuration/field'
+require 'solr/core_configuration/core'
+require 'solr/core_configuration/core_definition_builder'
 require 'solr/errors/solr_url_not_defined_error'
+require 'solr/errors/multiple_solr_cores_error'
 
 module Solr
   class Configuration
-    attr_accessor :faraday_options, :fields, :test_connection
-    attr_writer :url
+    attr_accessor :faraday_options, :cores, :test_connection
 
     def initialize
       @faraday_options = { request: { timeout: 2, open_timeout: 8 } }
-      @url = ENV['SOLR_URL']
-      @fields = {}
+      @cores = {}
     end
 
-    def uri
-      @uri ||= Addressable::URI.parse(url)
+    def uri(core_name: nil)
+      Addressable::URI.parse(url(core_name: core_name))
     end
 
-    def url
-      raise Errors::SolrUrlNotDefinedError unless @url
-      @url
+    def url=(value)
+      if value.nil?
+        raise Errors::SolrUrlNotDefinedError
+      else
+        @url = value
+      end
     end
 
-    def define_fields
-      builder = Solr::FieldConfiguration::FieldDefinitionBuilder.new
+    def url(core_name: nil)
+      core_url = cores[core_name.to_sym]&.url if core_name
+      core_url ||= @url
+      core_url ||= default_core.url
+      raise Errors::SolrUrlNotDefinedError unless core_url
+      core_url
+    end
+
+    def default_core_name
+      default_core.name
+    end
+
+    def default_core
+      raise Errors::MultipleSolrCoresError if cores.count > 1
```

Comment:
it's fine to have multiple cores, but in this situation it's ambiguous

Reply:
`default_core` shouldn't be called if there are more than 1 core defined

Reply:
Yes, your error name is incorrect. It should be AmbiguousCoreError because having multiple cores is not an error, having multiple cores and not selecting one is